akka-actor/src/main/scala/akka/actor/CoordinatedShutdown.scala

```diff
@@ -110,6 +109,9 @@ object CoordinatedShutdown extends ExtensionId[CoordinatedShutdown] with Extensi
     val coord = new CoordinatedShutdown(system, phases)
     initPhaseActorSystemTerminate(system, conf, coord)
     initJvmHook(system, conf, coord)
+    system.registerOnTermination {
+      if (!runningJvmHook) coord.removeJvmShutdownHooks()
+    }
```

Comment:
isn't the only problematic hook that we should take responsibility for removing the one that is added in `initJvmHook`?